src/rust/src/x509.rs

```diff
@@ -46,6 +83,115 @@ fn get_bit(input: &[u8], n: usize) -> bool {
     }
 }
 
+fn parse_name_attribute(
+    py: pyo3::Python<'_>,
+    attribute: AttributeTypeValue,
+) -> Result<pyo3::PyObject, PyAsn1Error> {
+    let x509_module = py.import("cryptography.x509")?;
+    let oid = x509_module
+        .call_method1("ObjectIdentifier", (attribute.type_id.to_string(),))?
+        .to_object(py);
+    let tag_enum = py
+        .import("cryptography.x509.name")?
+        .getattr("_ASN1_TYPE_TO_ENUM")?;
+    let py_tag = tag_enum.get_item(attribute.value.tag().to_object(py))?;
+    // TODO: is it safe to assume utf8? Current Python does.
+    let py_data =
+        std::str::from_utf8(attribute.value.data()).map_err(|_| asn1::ParseError::InvalidValue)?;
+    Ok(x509_module
+        .call_method1("NameAttribute", (oid, py_data, py_tag))?
+        .to_object(py))
+}
+
+fn parse_name(py: pyo3::Python<'_>, name: Name) -> Result<pyo3::PyObject, PyAsn1Error> {
+    let x509_module = py.import("cryptography.x509")?;
+    let py_list = pyo3::types::PyList::empty(py);
```

Comment:
Can we give these semantically meaningful names? Just calling itlist tells us nothing.